gnomad/utils/annotations.py

```diff
@@ -2312,21 +2312,45 @@ def _create_group_dicts(
         "cohort": {"id": "ALL"},
     }
 
-    # Create ancillaryResults for additional frequency and popMaxFAF95 information
+    # Check allele balance for heterozygotes values.
+    # Flagged allele balance values are those in bins > 0.90.
+    # Each bin is 0.05, so flagged values are in the last 2 bins.
+    if len(input_struct.ab_hist_alt.bin_freq) != 20:
+        raise ValueError(
+            f"{gnomad_id} ab_hist_alt.bin_freq had "
+            f"{len(input_struct.ab_hist_alt.bin_freq)} items, expected 20"
+        )
+    # The bin_freq should be in order but we can verify the order from bin_edges.
+    ab_bin_freq = list(
+        map(
+            lambda tpl: tpl[1],
+            sorted(
+                zip(
+                    input_struct.ab_hist_alt.bin_edges,
+                    input_struct.ab_hist_alt.bin_freq,
+                ),
+                key=lambda tpl: tpl[0],
+            ),
+        )
+    )
+
+    # Create ancillaryResults for additional frequency and popMaxFAF95 information.
     ancillaryResults = {
         "homozygotes": overall_freq["homozygote_count"],
         "popMaxFAF95": {
             "frequency": input_struct.faf95.popmax,
             "confidenceInterval": 0.95,
         },
+        "qcFilters": list(input_struct.filters),
+        "lowComplexityRegion": input_struct.region_flag.lcr,
+        "heterozygousAlleleBalanceFlagged": sum(ab_bin_freq[-2:]),
     }
 
     if input_struct.faf95.popmax_population is not None:
-        ancillaryResults["popMaxFAF95"][
-            "popFreqID"
-        ] = f"{gnomad_id}.{input_struct.faf95.popmax_population.upper()}"
+        popFreqId = f"{gnomad_id}.{input_struct.faf95.popmax_population.upper()}"
     else:
-        ancillaryResults["popMaxFAF95"]["popFreqID"] = None
+        popFreqId = f"{gnomad_id}.NO_COHORTS"
```

Comment:
looks like we're no longer going with the NO_COHORTS option so all of popMaxFAF95 should be null in these cases